EIPS/eip-2502.md

```diff
@@ -0,0 +1,462 @@
+            d = checkpoints[checkpoints.length-1];
+        } else {
+            // Lookup in array;
+            uint min = 0;
+            uint max = checkpoints.length-1;
+            while (max > min) {
+                uint mid = (max + min + 1) / 2;
+                if (checkpoints[mid].fromBlock <= _block) {
+                    min = mid;
+                } else {
+                    max = mid-1;
+                }
+            }
+            d = checkpoints[min];
+        }
+    }
+
+     /**
+     * @notice Reads `_who` configured delegation at block number `_block` in this level,
+     *         or from parent level if `_who` never defined/defined to parent address.
+     * @param _who What address to lookup.
+     * @param _block Block number of what height in history.
+     * @return The address `_who` chosen delegate to.
+     */
+    function findDelegatedToAt(
+        address _who,
+        uint _block
+    )
+        internal
+        view
+        returns (address directDelegate)
+    {
+        DelegateSet[] storage checkpoints = delegations[_who];
+
+        //In case there is no registry
+        if (checkpoints.length == 0) {
+            return (
+                address(parentDelegation) == address(0) ?
+                address(0) : parentDelegation.delegatedToAt(_who, _block)
+            );
+        }
+        return getMemoryAt(checkpoints, _block).to;
+    }
+}
+```
+
+
+## Rationale
+
+Democracies are seen as a multisig of infinite participants, which signatures weight are on their voting token and the threshold as usually 50%+1 or more.
+
+Being possible to approve something in qualified quorums becomes more difficult the wider the community of holders is, and approving very specific tasks can be complicated to evaluate for most of holders.
+
+Therefore the design was around solving the problem of defining delegates in a common but optionally specialized way building a trust network which can easily and carefully approve proposals. 
+
+The support for Circular Delegations is interesting to allow a trust bond between a closed community, where any of the community can have the voice of all if none others vote.
+
+Default Delegate is useful for DAO startups, which need a Controlled, yet Trustless method of developing the system. 
+
+
+## Backwards Compatibility
+
+N/A
+
+## Test Cases
+
+TBD
+
+## Implementation
+
+Status.im µTopic Democracy Delegation
+
+## Security Considerations
+
+### Bribing effects
+
+Bribing Token Holders is not considered a risk, because stakeholders ultimately are responsible by everything approved. 
+
+The bribing of delegates should be mitigated by a veto period.
+
+### Veto Period
+
+Democracies willing to use Liquid Democracy for approving in chain actions should consider implementing a Veto Period
+
+## References
+
+https://docs.google.com/presentation/d/1FJkaJlp_Fs0D5KHsn8oEgDOhg9QKiBwxHOvUL0dvC5I/edit#slide=id.g50bded6ab0_0_10
+
```

Comment:
EIPs don't have a References section, which is intentional.  EIPs should be able to stand on their own without external reference material.  If there is some critical content for the EIP that must be included (forever) then it should be added as an asset in `../assets/eip-2502/` and linked as part of one of the other sections.